dags/js/IO/file-utils.js

```diff
@@ -1,12 +1,11 @@
 const fs = require('fs');
+const util = require('util');
 
 
 module.exports.deleteFile = async (filePath) => {
-  if (fs.existsSync(filePath)) {
-    await fs.unlink(filePath, (error) => {
-      if (error) throw error;
-      console.log('successfully deleted ' + filePath);
-    });
+  if (await util.promisify(fs.exists)(filePath)) {
```

Comment:
Using `existsSync` will make the code more readable.

Reply:
For reference no one should be using `fs.exists` ... ref: https://nodejs.org/api/fs.html#fs_fs_exists_path_callback 

/cc @libero/node-js-reviewers

Reply:
Have just gone away from `existsSync` because it's sync. Will switch to `fs.access` then.

Reply:
Why is this such a problem inside an `async` function?

Reply:
I created a `gist` to help illustrate that its fine... https://gist.github.com/diversemix/60cfcfb4cd378744aa03004cc4799c42

Reply:
Forked the gist to demonstrate how the sync filesystem calls should block the whole process including all outstanding promises:
https://gist.github.com/giorgiosironi/034b62b7e4d11e9285007a9b0a1e9717

Reply:
That's because the function isn't marked as async ... try changing line 24...
from

Reply:
Live debugging session on the Gist, agreed to cut out `*sync()` calls in general from this project as problematic.